### DISCUSSION TITLE
Release v4.5.0

### DISCUSSION THREAD

**Discussion by evan-buss** [evan-buss/openbooks#117] [Announcements]
## Added
-  Use `--username/-u` flag to optionally specify the [UserAgent](https://en.wikipedia.org/wiki/Client-to-client_protocol#VERSION) reported to the IRC server. Default remains `OpenBooks v4.5.0`.

## Breaking 
- `--name/-n` flag **must** be specified when starting the application. OpenBooks will no longer generate a random `noun_adjective` username.
- Only a single connection to the IRC server will be made. Opening a second browser tab will show an error message.

<hr /><em>This discussion was created from the release <a href='https://github.com/evan-buss/openbooks/releases/tag/v4.5.0'>Release v4.5.0</a>.</em>

**Comment by parvbarot** [comment]
I'm having trouble finding books in the new version; it keeps searching but not finding anything

**Reply by CyBuzz** [reply]
I am having the same issue. i believe it is a firewall thing. I hooked my laptop direct to my modem and I was getting results but as soon as I went behind the firewall, no results...just spinning circle. Something changed in the app since my firewall rules have not changed in quite a while and OpenBooks worked great prior. I have not been able to find any information elsewhere on the internet that can help with firewall rules.

![image](https://user-images.githubusercontent.com/35875714/213526974-48dc80db-097e-4091-adb0-1f238a655244.png)
![image](https://user-images.githubusercontent.com/35875714/213527291-bf4fce5c-1b92-431c-8eb0-5ca8109e159c.png)

**Comment by tam481** [comment]
Are you responsible for releasing the Docker Hub containers?
If yes, please could you release the arm7 or arm64 images if you can.

Thank you in advance

**Reply by evan-buss** [reply]
ARM images are now up under the `latest` tag. There was a config issue with the automated build

**Comment by zenjubbers** [comment]
For those of you using docker compose, you can add the --name parameter using

command: "--name IRC-random-username"

in your docker-compose.yml file.

**Reply by parvbarot** [reply]
Now that I cannot connect to the server, I am attaching my container logs  


      
2023/01/12 05:34:43 SERVER: Base Path: /openbooks/
2023/01/12 05:34:43 SERVER: OpenBooks is listening on port 80
2023/01/12 05:34:43 SERVER: Download Directory: /books
2023/01/12 05:34:43 SERVER: Open http://localhost:80/openbooks/ in your browser.
2023/01/12 05:34:53 SERVER: Client connected from 192.168.0.112:36423
2023/01/12 05:34:53 CLIENT (ridhdbsu): New client created.
2023/01/12 05:34:53 CLIENT (ridhdbsu): CONNECT Message Received
2023/01/12 05:34:54 CLIENT (ridhdbsu): EOF
2023/01/12 05:35:13 CLIENT (ridhdbsu): SEARCH Message Received
2023/01/12 05:36:03 CLIENT (ridhdbsu): Connection Closed: websocket: close 1001 (going away)
2023/01/12 05:36:03 SERVER: Client connected from 192.168.0.112:36441
2023/01/12 05:36:03 CLIENT (ridhdbsu): New client created.
2023/01/12 05:36:03 CLIENT (ridhdbsu): CONNECT Message Received
2023/01/12 05:36:04 CLIENT (ridhdbsu): EOF
2023/01/12 05:37:23 CLIENT (ridhdbsu): Connection Closed: websocket: close 1001 (going away)
2023/01/12 05:37:23 SERVER: Client connected from 192.168.0.112:36476
2023/01/12 05:37:23 CLIENT (ridhdbsu): New client created.
2023/01/12 05:37:24 CLIENT (ridhdbsu): CONNECT Message Received
2023/01/12 05:37:35 CLIENT (ridhdbsu): EOF
2023/01/12 05:37:35 CLIENT (ridhdbsu): Connection Closed: websocket: close 1001 (going away)
2023/01/12 05:37:35 CLIENT (ridhdbsu): Error writing JSON to websocket: websocket: close

**Comment by tam481** [comment]
I think there is a problem with the latest build. At least the arm64 release. It doesn't seem to work on my RPi 4. It creates the container and starts it but shuts down immediately. It shows the container as "Exited" with no logs.

**Reply by evan-buss** [reply]
Are you specifying a --name? The server won't start without it.

**Reply by CyBuzz** [reply]
I had the same issue and added --name and it worked.

**Comment by NiteCrawla** [comment]
Think it's having issues with windows. I had it running about 3 weeks ago and then now it won't even run. I see it open cmd prompt but after that t won't even open the browser.

**Comment by bt4ibwem8** [comment]
Does Openbooks still work?

**Reply by kernelkaribou** [reply]
Yes it still works, just has a few odd behaviors and open issues that would be beneficial to address. However as it stands, it functions and I use it regularly.
